tests/integration/devfile/docker/cmd_docker_devfile_test_test.go

```diff
@@ -0,0 +1,189 @@
+
+			output := helper.CmdShouldFail("odo", "test", "--context", context)
+
+			Expect(output).To(ContainSubstring("'odo test' is not supported in devfile 1.0.0"))
+		})
+
+		It("should show error if no test group is defined", func() {
+			helper.CmdShouldPass("odo", "create", "nodejs", "--context", context, cmpName)
+
+			helper.CopyExample(filepath.Join("source", "devfiles", "nodejs", "project"), context)
+			helper.CopyExampleDevFile(filepath.Join("source", "devfiles", "nodejs", "devfile.yaml"), filepath.Join(context, "devfile.yaml"))
+			helper.CmdShouldPass("odo", "push", "--context", context)
+			output := helper.CmdShouldFail("odo", "test", "--context", context)
+
+			Expect(output).To(ContainSubstring("the command group of kind \"test\" is not found in the devfile"))
+		})
+
+		It("should show error if specify non-exist command", func() {
+			helper.CmdShouldPass("odo", "create", "nodejs", "--context", context, cmpName)
+
+			helper.CopyExample(filepath.Join("source", "devfiles", "nodejs", "project"), context)
+			helper.CopyExampleDevFile(filepath.Join("source", "devfiles", "nodejs", "devfile-with-testgroup.yaml"), filepath.Join(context, "devfile.yaml"))
+			helper.CmdShouldPass("odo", "push", "--context", context)
+			output := helper.CmdShouldFail("odo", "test", "--test-command", "invalidcmd", "--context", context)
+
+			Expect(output).To(ContainSubstring("not found in the devfile"))
+		})
+
+		It("should show error if command from another group", func() {
+			helper.CmdShouldPass("odo", "create", "nodejs", "--context", context, cmpName)
+
+			helper.CopyExample(filepath.Join("source", "devfiles", "nodejs", "project"), context)
+			helper.CopyExampleDevFile(filepath.Join("source", "devfiles", "nodejs", "devfile-with-testgroup.yaml"), filepath.Join(context, "devfile.yaml"))
+			helper.CmdShouldPass("odo", "push", "--context", context)
+			output := helper.CmdShouldFail("odo", "test", "--test-command", "devrun", "--context", context)
+
+			Expect(output).To(ContainSubstring("command devrun is of group run in devfile.yaml"))
+		})
+
+		It("should show error if devfile has multiple default test command", func() {
+			helper.CmdShouldPass("odo", "create", "nodejs", "--context", context, cmpName)
+			helper.CopyExample(filepath.Join("source", "devfiles", "nodejs", "project"), context)
+			helper.CopyExampleDevFile(filepath.Join("source", "devfiles", "nodejs", "devfile-with-testgroup.yaml"), filepath.Join(context, "devfile.yaml"))
+			helper.ReplaceString("devfile.yaml", "isDefault: true", "")
+			helper.CmdShouldPass("odo", "push", "--context", context)
+			output := helper.CmdShouldFail("odo", "test", "--context", context)
+			Expect(output).To(ContainSubstring("there should be exactly one default command for command group test, currently there is no default command"))
+		})
+
+		It("should show error if devfile has multiple default test command", func() {
```

Comment:
duplicate test as above?

Reply:
The above one should be with no default command. Need to rename the test. 
Thanks for catching this

Reply:
updated